addons/hr/models/res_users.py

```diff
@@ -200,7 +171,8 @@ def write(self, vals):
         hr_fields = {
             field_name: field
             for field_name, field in self._fields.items()
-            if field.related_field and field.related_field.model_name == 'hr.employee' and field_name in vals
+            if field_name in vals
+            if callable(field.compute) and 'employee_field' in field.compute.__name__
```

Comment:
If the goal is to verify that the field is a field produced by your `field_employee` factory, it might be wise to verify it more precisely than with just part of the name (which is relatively generic).

Using `__qualname__` would allow you to do this, or you could add a "flag" (attribute) to the compute method.

It is up to you to decide if this is necessary, it is just a suggestion.

Reply:
`field.compute.__qualname__ == 'field_employee.<locals>.compute_employee_field'`
Like that? I went with a simple solution, I can change to whatever you prefer.